utils/post_processing/backconvert_borealis.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2019 SuperDARN Canada, University of Saskatchewan
+# Author: Liam Graham
+
+#
+# backconvert_borealis.py
+# 2019-07-30
+# Command line tool for converting restructured and compressed
+# Borealis files back to their initial site format
+import deepdish as dd
+import numpy as np
+import os
+import sys
+import subprocess as sp
+import datetime
+import warnings
+
+
+def backconvert_pre_bfiq(data_record):
+	"""
+	Converts a restructured antennas iq file back to its
+	original site format
+	Args:
+		data_record (dict): An opened antennas_iq hdf5 file
+	"""
+	num_records = len(data_record["int_time"])
+	ts_dict = dict()
+	# get keys from first sequence timestamps
+	for rec, seq_ts in enumerate(data_record["sqn_timestamps"]):
+		# format dictionary key in the same way it is done
+		# in datawrite on site
+		sqn_dt_ts = datetime.datetime.utcfromtimestamp(seq_ts[0])
+		epoch = datetime.datetime.utcfromtimestamp(0)
+		key = str(int((sqn_dt_ts - epoch).total_seconds() * 1000))
+		
+		ts_dict[key] = dict()
+		for f in data_record:
+			if not type(data_record[f]) is np.ndarray:
+				ts_dict[key][f] = data_record[f]
+			else:
+				if np.shape(data_record[f])[0] == num_records:
+					# pass data fields that are written per record
+					pass
+				else:
+					ts_dict[key][f] = data_record[f]
+		# Handle per record fields
+		num_sequences = data_record["num_sequences"][rec]
+		ts_dict[key]["num_sequences"] = num_sequences
+		ts_dict[key]["int_time"] = data_record["int_time"][rec]
+		ts_dict[key]["sqn_timestamps"] = data_record["sqn_timestamps"][rec, 0:int(num_sequences)]
+		ts_dict[key]["noise_at_freq"] = data_record["noise_at_freq"][rec, 0:int(num_sequences)]
+		ts_dict[key]["data_descriptors"] = ts_dict[key]["data_descriptors"][1:]
+		ts_dict[key]["data_dimensions"] = data_record["data_dimensions"][rec]
+
+		ts_dict[key]["data"] = np.trim_zeros(data_record["data"][rec].flatten())
+	
+	return ts_dict
+
+def backconvert_bfiq(data_record):
```

Comment:
This method looks identical to the backconvert_pre_bfiq method. You could remove one of them I think.

Reply:
I hadn't noticed before, but you're right.